frontend/packages/ux-editor/src/components/config/editModal/EditOptions/OptionTabs/ReferenceTab/ReferenceTab.tsx

```diff
@@ -37,7 +54,12 @@ export function ReferenceTab({
         value={component.optionsId}
         size='small'
       />
-      <p style={{ marginBottom: 0 }}>
+      {(isOptionsIdInLibrary || component.options) && (
+        <StudioAlert severity={'info'} size='sm'>
+          {t('ux_editor.options.tab_referenceid_alert_title')}
```

Comment:
I think this alert will be visible also if one tries to enter an optionListId that exists in the library. Not sure how we should avoid that, so maybe it is easier to adapt the text (Which is not added in this PR 😬 ) to fit both cases 🤷

Reply:
The alert should be visible if the optionsId is from the library, is that not the point of it?

Reply:
Yes, but you can still enter a optionId from the library as your custom id. And then this alert will be shown. Given the original text; `Du har allerede referert til en kodeliste. Skriver du inn en ID, vil referansen bli slettet."` this scenario will not match what is happening. The correct case then would be something like: `ID-en du skrev inn tilhører allerede en eksisterende kodeliste i biblioteket.` 🫣

Reply:
oh... right. I'm unsure how we would know what the user is typing without keeping it in a local variable of a sort.

Reply:
Yup, I know, so maybe we should adapt the text in a way that makes it fit both scenarios 🤔 Or actually, I just tested and it seems like that scenario will be bigger thing to fix maybe. In staging, if typing a name of an existing code list, the whole EditTab is rerendered (which is understandable) and you get the view as if you have selected a code list from the library.

Reply:
Concerning the other issue; that the input field in the reference tab is filled in with the option list id from the library if you have set one, I imagined fixing with setting the value to something like this;
```
const referenceIdValue = optionListIds ? undefined : component.optionsId;
```
Not sure it will work. When I tested now I also discovered a new issue, which is also present in staging. The issue is that if you add an actual custom option list id, and go back to the `Velg kodeliste` tab you get this view:

![Skjermbilde 2024-12-18 kl  16 32 38](https://github.com/user-attachments/assets/d45b1c77-f092-403c-b0c8-c0643d7622ef)

It might be fixed in the other PR you are working on tho 🤷

Reply:
The missing content will be fixed by [this PR](https://github.com/Altinn/altinn-studio/pull/14276). I've added a line to display the referenceId only if it's not from the library.
```tsx
const referenceIdValue = isOptionsIdInLibrary ? undefined : component.optionsId;
```